src/Analyzers/CSharp/Analyzers/UseNullPropagation/CSharpUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -36,29 +40,66 @@ protected override bool IsInExpressionTree(SemanticModel semanticModel, SyntaxNo
             => node.IsInExpressionTree(semanticModel, expressionTypeOpt, cancellationToken);
 
         protected override bool TryAnalyzePatternCondition(
-            ISyntaxFacts syntaxFacts, SyntaxNode conditionNode,
-            [NotNullWhen(true)] out SyntaxNode? conditionPartToCheck, out bool isEquals)
+            ISyntaxFacts syntaxFacts, ExpressionSyntax conditionNode,
+            [NotNullWhen(true)] out ExpressionSyntax? conditionPartToCheck, out bool isEquals)
         {
             conditionPartToCheck = null;
             isEquals = true;
 
             if (conditionNode is not IsPatternExpressionSyntax patternExpression)
-            {
                 return false;
-            }
 
-            if (patternExpression.Pattern is not ConstantPatternSyntax constantPattern)
+            var pattern = patternExpression.Pattern;
+            if (pattern is UnaryPatternSyntax(SyntaxKind.NotPattern) notPattern)
             {
-                return false;
+                isEquals = false;
+                pattern = notPattern.Pattern;
             }
 
+            if (pattern is not ConstantPatternSyntax constantPattern)
+                return false;
+
             if (!syntaxFacts.IsNullLiteralExpression(constantPattern.Expression))
-            {
                 return false;
-            }
 
             conditionPartToCheck = patternExpression.Expression;
             return true;
         }
+
+        protected override bool TryGetPartsOfIfStatement(
+            IfStatementSyntax ifStatement,
+            [NotNullWhen(true)] out ExpressionSyntax? condition,
+            [NotNullWhen(true)] out StatementSyntax? trueStatement)
+        {
+            // has to be of the form:
+            //
+            //   if (...)
+            //      statement
+            //
+            // or
+            //
+            //   if (...)
+            //   {
+            //       statement
+            //   }
+
+            condition = ifStatement.Condition;
+
+            trueStatement = null;
+            if (ifStatement.Else == null)
```

Comment:
Consider inverting this and returning early. The VB version of this was nicer to read IMO